docs/storage/volume_migration.md

```diff
@@ -0,0 +1,219 @@
+      volumes:
+      - persistentVolumeClaim:
+          claimName: dest-pvc
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+
+The destination volume may be of a different type or size than the source. It is possible to migrate from and to a block volume as well as a filesystem volume.
+The destination volume should be equal to or larger than the source volume. However, the additional difference in the size of the destination volume is not instantly visible within the VM and must be manually resized because the guest is unaware of the migration.
+
+The volume migration depends on the `VolumeMigration` feature gate. To enable the feature gate, add the following to the KubeVirt CR:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+        - VMLiveUpdateFeatures
+        - VolumesUpdateStrategy
+        - VolumeMigration
+  workloadUpdateStrategy:
+    workloadUpdateMethods:
+    - LiveMigrate
+```
+
+## Datavolume template
+
+Updating a datavolume that is referenced by a datavolume template requires special caution. The volumes section must include a reference to the name of the datavolume template. This means that the datavolume templates must either be entirely deleted or updated as well.
+
+Example of updating the datavolume for the original VM in the first example:
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: dst-dv
+spec:
+  source:
+      blank: {}
+  pvc:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+    storageClassName: local
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-dv
+  name: vm-dv
+spec:
+  updateVolumesStrategy: Migration
+  dataVolumeTemplates:
+  - metadata:
+      name: dst-dv
+    spec:
+      pvc:
+        accessModes:
+        - ReadWriteOnce
+        volumeMode: Filesystem
+        resources:
+          requests:
+            storage: 2Gi
+        storageClassName: local
+      source:
+        registry:
+          url: docker://registry:5000/kubevirt/alpine-container-disk-demo:devel
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-dv
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: dst-dv
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+## Limitations
+
+Only certain types of disks and volumes are supported to be migrated. For an invalid type of volume the RestartRequired condition is set and volumes will be replaced upon VM restart.
```

Comment:
> For an invalid type of volume the RestartRequired condition is set and volumes will be replaced upon VM restart.

So if the storage type is valid - migration will happen right after changing the VM yaml, and the restart is required?

Reply:
Correct, if the update of the volume is valid then the migration is automatically triggered, otherwise, the Restart Required condition is set.

Reply:
Got it, thanks